ansible_collections/arista/avd/roles/eos_validate_state/tasks/main.yml

```diff
@@ -0,0 +1,281 @@
+        (ethernet_interface.value.peer_interface is defined and ethernet_interface.value.peer_interface is not none) and
+        (hostvars[ethernet_interface.value.peer] is defined and hostvars[ethernet_interface.value.peer] is not none) and
+        (hostvars[ethernet_interface.value.peer]['dns_domain'] is defined and hostvars[ethernet_interface.value.peer]['dns_domain'] is not none)
+      ignore_errors: "{{ validation_mode_loose }}"
+      register: lldp_topology_results
+      tags:
+        - lldp_topology
+
+    - name: validate MLAG status
+      eos_command:
+        commands: "show mlag detail | json"
+        waitfor:
+          - "result[0].state eq 'active'"
+          - "result[0].negStatus eq 'connected'"
+        retries: 2
+        interval: 2
+      ignore_errors: "{{ validation_mode_loose }}"
+      when: |
+        (mlag_configuration is defined and mlag_configuration is not none)
+      tags:
+        - avd
+
+    - name: Gather IP reachability state (directly connected interfaces)
+      eos_command:
+        commands: "ping {{ hostvars[ethernet_interface.value.peer]['ethernet_interfaces'][ethernet_interface.value.peer_interface]['ip_address'] | ipaddr('address')}} source {{ ethernet_interface.value.ip_address  | ipaddr('address')}} repeat 1"
+      loop: "{{ethernet_interfaces | default({}) | dict2items}}"
+      loop_control:
+        loop_var: ethernet_interface
+      ignore_errors: "{{ validation_mode_loose }}"
+      when: |
+        (ethernet_interfaces is defined and ethernet_interfaces is not none) and
+        (ethernet_interface.value.type is defined and ethernet_interface.value.type is not none and ethernet_interface.value.type == 'routed') and
+        (ethernet_interface.value.ip_address is defined and ethernet_interface.value.ip_address is not none) and
+        (ethernet_interface.value.peer is defined and ethernet_interface.value.peer is not none) and
+        (ethernet_interface.value.peer_interface is defined and ethernet_interface.value.peer_interface is not none) and
+        (hostvars[ethernet_interface.value.peer]['ethernet_interfaces'][ethernet_interface.value.peer_interface]['ip_address'] is defined and
+        hostvars[ethernet_interface.value.peer]['ethernet_interfaces'][ethernet_interface.value.peer_interface]['ip_address'] is not none)
+      register: ip_reachability_state
+      tags:
+        - ip_reachability
+
+    - name: validate ip reachability (directly connected interfaces)
+      assert:
+        that:
+          - ip_reachability_test.stdout[0] | regex_search("1 received")
+        quiet: true
+      loop: "{{ ip_reachability_state.results }}"
+      loop_control:
+        loop_var: ip_reachability_test
+      ignore_errors: "{{ validation_mode_loose }}"
+      when: ( ip_reachability_test.skipped is not defined )
+      register: ip_reachability_results
+      tags:
+        - ip_reachability
```

Comment:
Maybe we could reduce number of tags by using more generic name like:

- __topology__ to cover `ip_reachability` and `lldp_topology`
- __bgp__ to cover `ArBGP_state`, `bgp_**`

Reply:
Agreed, what about blending this approach, as we can have multiple tags on a task. This way a group of test or specific tests can be run.